tests/ha/single_pod_restart/test_data_pod_restart_api.py

```diff
@@ -1221,3 +1220,307 @@ def test_obj_ver_suspension_data_pod_restart(self):
+                                       clstr_status=True)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], f"Failed to restore pod by {self.restore_method} way")
+        self.restore_pod = False
+        event.clear()
+        put_thread.join()
+        get_thread.join()
+        LOGGER.info("Step 6: Successfully restart data pod with replica method & checked "
+                    "cluster status.")
+
+        LOGGER.info("Step 7: Verify background Put & Get Version for %s", self.object_name)
+        get_resp = tuple()
+        put_resp = tuple()
+        while len(get_resp) != 2:
+            get_resp = get_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        while len(put_resp) != 2:
+            put_resp = put_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        assert_utils.assert_false(len(get_resp[1]) or len(put_resp[1]),
+                                  "Failure in put or get object during restart"
+                                  f"GET resp: {get_resp[1]} PUT resp: {put_resp[1]}")
+        self.version_etag[self.bucket_name].extend(put_resp[0])
+        LOGGER.info("Step 7: Verified background Put & Get Version for %s", self.object_name)
+
+        LOGGER.info("Step 8: Get object with version IDs & verify etags for %s.", self.bucket_name)
+        resp = self.ha_api.parallel_get_object(event=event, s3_ver_obj=self.s3_ver,
+                                               bkt_name=self.bucket_name, obj_name=self.object_name,
+                                               ver_etag=self.version_etag[self.bucket_name])
+        assert_utils.assert_true(resp[0], f"Get Object with versionID failed {resp[1]}")
+        LOGGER.info("Step 8: Got object with version IDs & verified etags for %s.",
+                    self.bucket_name)
+
+        if CMN_CFG["dtm0_disabled"]:
+            new_bucket = f"ha-mp-bkt-{int(perf_counter_ns())}"
+            download_path = os.path.join(self.test_dir_path, self.test_file + "_new")
+            self.extra_files.append(download_path)
+            LOGGER.info("Step 9: Create new bucket and upload object %s of %s size. Enable "
+                        "versioning on %s.", self.object_name, self.f_size, new_bucket)
+            args = {'chk_null_version': True, 'is_unversioned': True, 'file_path': download_path,
+                    'enable_ver': True, 's3_ver': self.s3_ver}
+            resp = self.ha_api.crt_bkt_put_obj_enbl_ver(event, self.s3_test_obj, new_bucket,
+                                                        self.object_name, **args)
+            assert_utils.assert_true(resp[0], resp[1])
+            self.version_etag.update({new_bucket: []})
+            self.version_etag[new_bucket].extend(resp[1])
+            LOGGER.info("Step 9: Created bucket and uploaded object %s of %s size. Enabled "
+                        "versioning on %s.", self.object_name, self.f_size, new_bucket)
```

Comment:
Should we try uploading same object to existing bucket?
You can keep steps 10 and 11 common for dtm enabled/disabled case.
Keep only step 9 inside dtm flag condition.
That way we can check if versioning works after restart also in case of dtm enabled.